Inventory.cs

```diff
@@ -0,0 +1,170 @@
+        {
+        }
+    }
+    public class Inventory
+    {
+        private List<Product> Products  = new List<Product>();
+        public void AddProduct()
+        {
+            try {
+                Console.Write("Enter the name of the product: ");
+                string name = Console.ReadLine();
+
+                if (string.IsNullOrWhiteSpace(name))
+                {
+                    throw new ProductsException("Product name cannot be empty");
+                }
+
+                Console.Write("Enter the price of the product: ");
+                double price;
+                while (!double.TryParse(Console.ReadLine(), out price))
+                {
+                    Console.WriteLine("Invalid input! Please enter a valid price.");
+                }
+
+                Console.Write("Enter the quantity of the product: ");
+                int quantity;
+                while (!int.TryParse(Console.ReadLine(), out quantity))
+                {
+                    Console.WriteLine("Invalid input! Please enter a valid quantity.");
+                }
+
+                Products.Add(new Product(name, price, quantity));
+                Console.WriteLine($"Whooooh! A new Product added with name: {name}, price: {price}, and quantity: {quantity}");
+            }
+            catch (ProductsException e)
+            {
+                Console.WriteLine(e.Message);
+            }
+            }
+        public void ViewAllProducts()
+        {
+            try
+            {
+                int Size = Products.Count;
+                if (Size > 0)
+                {
+                    Console.WriteLine($"\n We have {Size} products in our inventory\n");
+                    Console.WriteLine("--------------------------------------------------------------------");
+                    Console.WriteLine("|       Name       |       Price       |       Quantity       |");
+                    Console.WriteLine("--------------------------------------------------------------------");
+                    foreach (Product product in Products)
+                    {
+                        Console.WriteLine($"| {product.Name,-15} | {product.Price,-17} | {product.QuantityInStock,-20} |");
+                    }
+                    Console.WriteLine("--------------------------------------------------------------------");
+                }
+                throw new ProductsException("There is no product to view.");
+            }
+            catch (ProductsException ex) 
+            {
+                Console.WriteLine(ex.Message);
+            }
+        }
+
+        public void EditProduct()
+        {
+            try {
+                Console.Write("Enter the name of the product to edit: ");
+                string? name = Console.ReadLine();
+
+                foreach (Product product in Products)
+                {
+                    if (product.Name.Equals(name, StringComparison.Ordinal))
```

Comment:
Ordinal is for string cultures, for case sensitive, using OrdinalIgnoreCase
Read more in this article
https://learn.microsoft.com/en-us/dotnet/api/system.stringcomparison?view=net-8.0#fields